client/src/main/java/io/pravega/client/stream/impl/TransactionalEventStreamWriterImpl.java

```diff
@@ -144,15 +144,15 @@ public void abort() {
                         log.debug("Got exception while writing to transaction on abort: {}", e.getMessage());
                     }
                 }
-                getAndHandleExceptions(controller.abortTransaction(stream, txId), RuntimeException::new);
+                getAndHandleExceptions(controller.abortTransaction(stream, txId), TxnFailedException::new);
```

Comment:
I think what we really want if for the controller client to dictate the exception type. 
So instead of converting this, I think we should instead call `getThrowingException` which will just pass up whatever the underlying error was.